src/test/java/TestAhoCorasickDoubleArrayTrie.java

```diff
@@ -101,6 +102,51 @@ public void hit(int begin, int end, String value)
         });
     }
 
+    private static class CountHits implements AhoCorasickDoubleArrayTrie.IHitCancellable<String> {
+        private int count = 0;
+        private boolean countAll = true;
+
+        public CountHits(boolean countAll) {
+            this.countAll = countAll;
+        }
+
+        public int getCount() {
+            return count;
+        }
+
+        @Override
+        public boolean hit(int begin, int end, String value) {
+            count += 1;
+            return countAll;
+        }
+    }
+
+    public void testCancellation() throws Exception {
+        // Collect test data set
+        TreeMap<String, String> map = new TreeMap<String, String>();
```

Comment:
``` diff
- new TreeMap<String, String>();
+ new TreeMap<>();
`

Reply:
Not possible since it's marked to be compatible with Java 1.6 in the pom file.

Reply:
Right, forget my comments involving Java 8 features then :+1: